Firmware/MotorControl/axis.cpp

```diff
@@ -397,29 +397,29 @@ bool Axis::run_homing() {
 
     error_ &= ~ERROR_MIN_ENDSTOP_PRESSED; // clear this error since we deliberately drove into the endstop
 
-    // pos_setpoint is the starting position for the trap_traj so we need to set it.
-    controller_.pos_setpoint_ = min_endstop_.config_.offset;
-    controller_.vel_setpoint_ = 0.0f;  // Change directions without decelerating
-
-    // Set our current position in encoder counts to make control more logical
-    encoder_.set_linear_count((int32_t)(controller_.pos_setpoint_ * encoder_.config_.cpr));
-
+    // Calculate the desired position after offset.
+    float inputBuffer = encoder_.pos_estimate_.any().value() + min_endstop_.config_.offset;
```

Comment:
We need to check if `encoder_.pos_estimate_.any()` is `std::nullopt`. If so, `.value()` shouldn't be called and we need to abort homing.
`inputBuffer` => `input_buffer`

Reply:
Added an error for the abort case, and fixes my naming convention. Too much Java lately...

See latest [commit](https://github.com/odriverobotics/ODrive/pull/550/commits/d68dfbda824ffb336dccf5eb3ee2c602998c7551).